examples/abitest/module_0/rust/src/lib.rs

```diff
@@ -479,35 +497,40 @@ impl FrontendNode {
     }
 
     fn test_channel_wait_raw(&self) -> std::io::Result<()> {
-        let (out_channel, in_channel) = oak::channel_create().unwrap();
-
-        // Write a message to the channel so wait operations don't block.
-        let data = vec![0x01, 0x02, 0x03];
-        expect_eq!(OakStatus::OK, oak::channel_write(out_channel, &data, &[]));
+        let (out_channel, in_channel, _) = channel_create_raw();
 
         unsafe {
+            // Write a message to the channel so wait operations don't block.
+            let data = vec![0x01, 0x02, 0x03];
+            expect_eq!(
+                OakStatus::OK.value() as u32,
+                oak::wasm::channel_write(
+                    out_channel,
+                    data.as_ptr(),
+                    data.len(),
+                    &[] as *const u8,
+                    0,
+                )
+            );
+
             expect_eq!(
                 OakStatus::ERR_INVALID_ARGS.value() as u32,
                 oak::wasm::wait_on_channels(invalid_raw_offset() as *mut u8, 1)
             );
         }
 
         unsafe {
-            // Wait on [write handle, ready read handle, empty read handle].
-            const COUNT: usize = 3;
+            // Wait on [write handle, ready read handle].
+            const COUNT: usize = 2;
             let mut space = Vec::with_capacity(COUNT * oak::wasm::SPACE_BYTES_PER_HANDLE);
             space
-                .write_u64::<byteorder::LittleEndian>(out_channel.handle)
+                .write_u64::<byteorder::LittleEndian>(out_channel)
                 .unwrap();
             space.push(0x00);
             space
-                .write_u64::<byteorder::LittleEndian>(in_channel.handle)
+                .write_u64::<byteorder::LittleEndian>(in_channel)
                 .unwrap();
             space.push(0x00);
-            space
-                .write_u64::<byteorder::LittleEndian>(self.backend_in[0].handle)
-                .unwrap();
-            space.push(0x99); // deliberate nonsense status value
```

Comment:
Would still be nice not to lose the `NOT_READY` subcase.

Reply:
I've added a TODO, so this PR isn't held up longer